### PR TITLE
102203 about the school

### PR DESCRIPTION
**Pull request by MarkRobinsonDfe**
flesh out HTML as per ticket:-
https://dfe-gov-uk.visualstudio.com.mcas.ms/Academies-and-Free-Schools-SIP/_workitems/edit/102203?McasTsid=26110

Had to make some model property changes now I know that schoolId is really URN and now the trams API call is expecting a real URN value

Dev test URL is:-
https://localhost:44350/school/school-overview?appId=2147483647&urn=101934

### CODE REVIEW

File: Dfe.Academies.External.Web/Services/ConversionApplicationRetrievalService.cs [hunk 1]
```diff
@@ -99,20 +99,27 @@ public async Task<List<ConversionApplicationAuditEntry>> GetConversionApplicatio
...
         List<ConversionApplicationComponent> conversionApplicationComponents = new()
         {
-            new(name:"Contact details") {Id = 1, SchoolId = schoolId, Status = Status.Completed},
-            new(name:"Performance and safeguarding") {Id = 2, SchoolId = schoolId, Status = Status.InProgress},
-            new(name:"Pupil numbers") {Id = 3, SchoolId = schoolId, Status = Status.NotStarted},
-            new(name:"Finances") {Id = 4, SchoolId = schoolId, Status = Status.NotStarted},
-            new(name:"Partnerships and affiliations") {Id = 5, SchoolId = schoolId, Status = Status.NotStarted},
-            new(name:"Religious education") {Id = 6, SchoolId = schoolId, Status = Status.NotStarted},
+            //V1:-
+            new(name:"About the conversion") {Id = 1, SchoolId = schoolId, Status = Status.InProgress},
```
Comment (CathLass) [inline Dfe.Academies.External.Web/Services/ConversionApplicationRetrievalService.cs:109]: will SchoolId be urn now, in line with other changes you've made?

File: Dfe.Academies.External.Web/Controllers/BaseController.cs [hunk 2]
```diff
@@ -8,44 +8,30 @@ public class BaseController : Controller
...
-
-        protected async Task<ApplicationCacheValuesViewModel> LoadAndSetApplicationDetails(int applicationId)
-        {
-            ApplicationCacheValuesViewModel cachedValuesViewModel = null;
-            var applicationDetails = await _conversionApplicationRetrievalService.GetApplication(applicationId, Enums.ApplicationTypes.FormNewMat);
+            ApplicationCacheValuesViewModel? cachedValuesViewModel = null;
+            var applicationDetails = await ConversionApplicationRetrievalService.GetApplication(applicationId, Enums.ApplicationTypes.FormNewMat);
 
-            if (applicationDetails != null)
+            if (Object.ReferenceEquals(applicationDetails, null))
```
Comment (CathLass) [inline Dfe.Academies.External.Web/Controllers/BaseController.cs:28]: Is this change to get rid of a warning?

**Review by CathLass — APPROVED** [review]
I notice some method signatures have changed but there are no test changes - can any of these methods have unit tests added, e.g. `LoadAndSetSchoolDetails(int applicationId, int urn)`

File: Dfe.Academies.External.Web/Services/ConversionApplicationRetrievalService.cs (hunk 1 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web/Services/ConversionApplicationRetrievalService.cs:109]: I need URN to do the API lookup from trams which is why i've switched schoolId -> urn so it makes more sense when looking at the code

File: Dfe.Academies.External.Web/Controllers/BaseController.cs (hunk 2 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web/Controllers/BaseController.cs:28]: Yes, this removes a null reference exception warning

**Pull request merged**